Port hotbar to react-lua

coolpeter98 opened the pull request.

coolpeter98 committed:
Add hotbar

coolpeter98 committed:
Remove selene.toml

EliTheGingerCat commented on src/ReplicatedStorage/client/visuals/hotbar.luau:
```diff
@@ -0,0 +1,248 @@
+			},
+				create_element("UIGradient",{
+					Color = ColorSequence.new({
+						ColorSequenceKeypoint.new(0, Color3.fromRGB(201, 201, 201)),
+						ColorSequenceKeypoint.new(1, Color3.fromRGB(70, 70, 70)),
```
Personally I prefer using `Color3.new`, but I guess this is an inconsistency that can be tolerated in the code base.

This is not supposed to be a requested change but whatever you can just close this.

EliTheGingerCat commented on src/ReplicatedStorage/client/visuals/hotbar.luau:
```diff
@@ -0,0 +1,248 @@
+					Rotation = -90
+				})
+			),
+			create_element("Frame",{ --health bar
```
The health bar should *definitely* use a property.

EliTheGingerCat commented on src/ReplicatedStorage/client/visuals/hotbar.luau:
```diff
@@ -0,0 +1,248 @@
+					})
+				)
+			),
+			create_element("Frame",{ --absorption bar
```
Are we sure we want to add absorption / extra health? This is a very Mincraftian concept.

coolpeter98 replied:
Yes.

EliTheGingerCat commented on src/ReplicatedStorage/client/visuals/hotbar.luau:
```diff
@@ -0,0 +1,248 @@
+				HorizontalAlignment = "Center",
+				SortOrder = "LayoutOrder",
+			}),
+--[[
+
+this will be extremely messy
+i'm not sure how to make the equivalent of functions in react
+
+]]
```
Okay bro pack it up. You can just make a loop at the beginning of this function (`main`) and make an array, then stick that array in the child parameter of whatever `create_element` call it should be under.

To make a component, simply create another file that returns a function which accepts properties and returns an Instance. In fact, this function, `main`, is a component!

EliTheGingerCat reviewed (changes requested):
Also follow other general suggestions from other code reviews, namely naming the function correctly (bro really said "namely naming" 💀) and style (I did not see any styling issues this time but just a general suggestion ✅).

Next time I will review pull requests in order of creation so that, if any review references previous context, you will have already read it. I *could* link the comments here but I think that does not provide much value in this case, especially considering I will create a style document (maybe we should use [StyLua](<https://github.com/JohnnyMorganz/StyLua>)?).

coolpeter98 committed:
Make styling more consistent

coolpeter98 committed:
Use Color3.fromRGB instead of Color3.new

EliTheGingerCat committed:
refactor: split hotbar into smaller components

Also add styling changes.

EliTheGingerCat committed:
feat: make main-health display be functional

EliTheGingerCat committed:
Merge branch 'develop' into coolpeter98-react-lua-port-hotbar

EliTheGingerCat committed:
feat: make slots be selectable using keys

EliTheGingerCat committed:
feat: add RbxUtil Signal

EliTheGingerCat committed:
build: add option to build without map

This will make using the plugin a lot faster since it will not have to
deal with the whole map every time.

This relies on commit:
EliTheGingerCat/roblox-build-tools@d70674b1e048335ce457577e5d902eda7995c79a

EliTheGingerCat committed:
build: add serving, with whole map and without

This relies on commit:
EliTheGingerCat/roblox-build-tools@b43489acc27decb6e1cc3d2952ab886eee01e6cb

EliTheGingerCat committed:
fix: fix syncback script due to breaking change

This relies on commit:
EliTheGingerCat/roblox-build-tools@e8d6b9a984a4914cac6b33e15a47c08da119ce63

EliTheGingerCat committed:
build: add sourcemap script

This relies on commit:
EliTheGingerCat/roblox-build-tools@94c8e2579bf060da4585bd4ace5c6e3ab6b0314a

EliTheGingerCat committed:
feat: sync selected tool with hotbar interface

EliTheGingerCat committed:
docs: use `Color3.new`

EliTheGingerCat committed:
feat: add dynamic scaling for hotbar

EliTheGingerCat committed:
fix: anchor platform

EliTheGingerCat committed:
fix: set extra health to 0

Extra health will be implemented in a future pull request.
